### PR TITLE
ncgopher: Init at 0.1.5

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/87784

###### Things done
Packaged ncgopher at release 0.1.5. (Spoiler: takes some time to build)

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OmnipotentEntity** [comment]
The cargo sha256 is not matching for me.

```
[14.5 MiB DL]
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/96372/head:refs/nixpkgs-review/1
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 16 (delta 6), reused 6 (delta 6), pack-reused 3
Unpacking objects: 100% (16/16), 189.59 KiB | 1.97 MiB/s, done.
From https://github.com/NixOS/nixpkgs
   232dbcc5a13..93b5c08d48f  master     -> refs/nixpkgs-review/0
$ git worktree add /root/.cache/nixpkgs-review/pr-96372/nixpkgs 93b5c08d48f4ef9d6b700301873e6673d20a1170
Preparing worktree (detached HEAD 93b5c08d48f)
Updating files: 100% (22574/22574), done.
HEAD is now at 93b5c08d48f Merge pull request #86449 from sorki/dt-schema
$ git merge --no-commit 7a055c11e4fbace9797b72c81779f74208ec10cc
Auto-merging pkgs/top-level/all-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /root/.cache/nixpkgs-review/pr-96372/build.nix
hash mismatch in fixed-output derivation '/nix/store/xgplx03hv5v1d15dmqi2p11wrqdzcl5a-ncgopher-0.1.5-vendor.tar.gz':
  wanted: sha256:11gn31jfdbjcp3mcg1c3iyxk3s0dkmiwa64q01zkvd9yw6n9yjv5
  got:    sha256:12r4vgrg2bkr3p61yxcsg02kppg84vn956l0v1vb08i94rxzc8zk
cannot build derivation '/nix/store/8r92zhm8qiqsph172c4r33bj2l35k6zi-ncgopher-0.1.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/504z6lj2zjsp959j5xnhy5j8pbkw2rz7-env.drv': 1 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/504z6lj2zjsp959j5xnhy5j8pbkw2rz7-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/96372
1 package failed to build:
ncgopher

[0.0 MiB DL]
error: build log of '/nix/store/8r92zhm8qiqsph172c4r33bj2l35k6zi-ncgopher-0.1.5.drv' is not available
$ nix-shell /root/.cache/nixpkgs-review/pr-96372/shell.nix
```

**Comment by SCOTT-HAMILTON** [comment]
All checked have passed, how is this possible?

**Comment by SCOTT-HAMILTON** [comment]
*checks

**Comment by jonringer** [comment]
all checks did not pass https://github.com/NixOS/nixpkgs/pull/96372/checks?check_run_id=1031772020

**Comment by SCOTT-HAMILTON** [comment]
changed vendorSha256 to ofborg's one, should if it.

File: pkgs/applications/networking/ncgopher/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  cargoSha256 = "12r4vgrg2bkr3p61yxcsg02kppg84vn956l0v1vb08i94rxzc8zk";
+  verifyCargoDeps = true;
+  
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    (lib.getDev ncurses6)
+    openssl
+    sqlite
+  ];
```
Comment (jonringer) [inline pkgs/applications/networking/ncgopher/default.nix:29]: unneeded, there's also some extraneous whitespace
```suggestion
  cargoSha256 = "12r4vgrg2bkr3p61yxcsg02kppg84vn956l0v1vb08i94rxzc8zk";

  nativeBuildInputs = [ pkg-config ];
  buildInputs = [
    ncurses6
    openssl
    sqlite
  ];
```

**Comment by OmnipotentEntity** [comment]
Builds fine.  Executable runs.  Attempting to browse gopherpedia I get an error when attempting to follow a link, but entering the url directly works fine.  I do not know if this is a bug with the program or a bug with the build.  But considering this is 0.1.5 it's probably a program bug.  So lgtm

**Comment by OmnipotentEntity** [comment]
For posterity the error message is:

```
i   ____                        _	null	(FALSE)	0
i  / __ \                      | |	null	(FALSE)	0
i | |  | | ___   ___  _ __  ___| |	null	(FALSE)	0
i | |  | |/ _ \ / _ \| '_ \/ __| |	null	(FALSE)	0
i | |__| | (_) | (_) | |_) \__ \_|	null	(FALSE)	0
i  \____/ \___/ \___/| .__/|___(_)	null	(FALSE)	0
i                    | |	null	(FALSE)	0
i                    |_|	null	(FALSE)	0
i	null	(FALSE)	0
iLooks like something went wrong with that request	null	(FALSE)	0
i	null	(FALSE)	0
i	null	(FALSE)	0
3Error 400	null	(FALSE)	0
i	null	(FALSE)	0
i	null	(FALSE)	0
1back to gopherpedia	/	gopherpedia.com	70

.
```

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/96372
1 package built:
ncgopher
```

**Pull request merged**